gclient/src/api/rpc.rs

```diff
@@ -95,4 +99,134 @@ impl GearApi {
             .await
             .map_err(Into::into)
     }
+
+    /// Read the program's state as a byte vector.
+    pub async fn read_state_bytes(
+        &self,
+        program_id: ProgramId,
+        at: Option<H256>,
+    ) -> Result<Vec<u8>> {
+        self.0
+            .rpc()
+            .request("gear_readState", rpc_params![H256(program_id.into()), at])
+            .await
+            .map_err(Into::into)
+    }
+
+    /// Read the program's state as decoded data.
+    pub async fn read_state<D: Decode>(
```

Comment:
How do you feel about separate these fns with and without non-optional "at" argument?

Reply:
I agree, but should we rework other functions too? 

For example, `read_state_at(..., at)` and `read_state(...)`?

Reply:
> I agree, but should we rework other functions too?
> 
> For example, `read_state_at(..., at)` and `read_state(...)`?

yeah, this way

Reply:
probably some smart wrapper that accepts closure would be good

Reply:
I've not found a reason to use a closure there. I've just split functions into two options.